sway/commands/move.c

```diff
@@ -507,9 +509,14 @@ static struct cmd_results *cmd_move_container(bool no_auto_back_and_forth,
 		return cmd_results_new(CMD_INVALID, expected_syntax);
 	}
 
-	if (destination->type == N_CONTAINER &&
-			container_is_scratchpad_hidden(destination->sway_container)) {
-		return cmd_move_to_scratchpad();
+	if (destination->type == N_CONTAINER) {
+		if (container_is_scratchpad_hidden(destination->sway_container)) {
+			return cmd_move_to_scratchpad();
+		}
+		if (container_is_scratchpad_hidden_or_child(destination->sway_container)) {
+			container_move_to_container(container, destination->sway_container);
```

Comment:
This isn't sufficient. It does prevent the segfault, but the workspace is not arranged so there is a gap where the container was

Reply:
Yeah I don't know what I was thinking here... New version handles it better hopefully, unrelated to the garbage in the first patch.